mmeval/metrics/dota_map.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from multiprocessing.pool import Pool
+from typing import Dict, List, Optional, Sequence, Tuple
+
+from mmeval.metrics.voc_map import VOCMeanAP
+from .utils.bbox_iou_rotated import (bbox_iou_rotated,
+                                     calculate_bboxes_area_rotated)
+
+
+def filter_by_bboxes_area_rotated(bboxes: np.ndarray,
+                                  min_area: Optional[float],
+                                  max_area: Optional[float]):
+    """Filter the rotated bboxes with an area range.
+
+    Args:
+        bboxes (numpy.ndarray): The bboxes with shape (n, 5) in 'xywha' format.
+        min_area (Optional[float]): The minimum area. If None, does not filter
+            the minimum area.
+        max_area (Optional[float]): The maximum area. If None, does not filter
+            the maximum area.
+    Returns:
+        numpy.ndarray: A mask of ``bboxes`` identify which bbox are filtered.
+    """
+    bboxes_area = calculate_bboxes_area_rotated(bboxes)
+    area_mask = np.ones_like(bboxes_area, dtype=bool)
+    if min_area is not None:
+        area_mask &= (bboxes_area >= min_area)
+    if max_area is not None:
+        area_mask &= (bboxes_area < max_area)
+    return area_mask
+
+
+class DOTAMeanAP(VOCMeanAP):
+    """DOTA evaluation metric.
+    This metric computes the DOTA mAP (mean Average Precision) with the given
+    IoU thresholds and scale ranges.
+    Args:
```

Comment:
About the docstring:

1. Need add blank line for each section or block. Please reference to the docstring of `VOCMeanAP`.
2. We should add a brief introduction of DOTA dataset and add the official link: https://captain-whu.github.io/DOTA/dataset.html